### PR TITLE
Annot in progress

### PR DESCRIPTION
**Pull request by tnkhanh**

### CODE REVIEW

File: src/frontend/discover.ml [hunk 1]
```diff
@@ -108,7 +108,7 @@ let compile_input_file (filename: string) : CI.program =
   let input_type = get_input_type filename in
   match input_type with
   | InpSepLogic -> filename |> SE.compile_sep_logic |> CI.mk_seplogic_prog
-  | InpBitcode -> filename |> LC.compile_bitcode |> CI.mk_llvm_prog
+  | InpBitcode -> filename |> (LC.compile_bitcode []) |> CI.mk_llvm_prog
```
Comment (taquangtrung) [inline src/frontend/discover.ml:111]: Can remove parentheses `(` `)`

File: src/llvm/ann.ml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+type value = [
+    | `Bug_start of string
```
Comment (taquangtrung) [inline src/llvm/ann.ml:2]: 1. Can remove `[` `]`

2. Use variant (`Bug_start`), which is simpler thann polymorphic variant  (`` `Bug_start``)
    See more here: https://dev.realworldocaml.org/variants.html

3.  Declare bug types similar to https://github.com/sbip-sg/discover-analyzer/blob/bc92b1d18c6c3aeca18dadba7347f787ecc7f1d7/src/common/bug.ml#L24 (or can reuse this module) and change the code to `Bug_start of bug_type list`

File: src/frontend/discover.ml (hunk 1 above)
Comment (tnkhanh) [reply src/frontend/discover.ml:111]: `(` `)` are removed

**Review by taquangtrung — COMMENTED** [review]
@tnkhanh: can you merge the newest code code from the `develop` branch to your branch first?
I'll merge this PR to ensure the code are updated.

We can open another PR to fix the issue with choosing instructions.

**Comment by tnkhanh** [comment]
> @tnkhanh: can you merge the newest code code from the `develop` branch to your branch first? I'll merge this PR to ensure the code are updated.
> 
> We can open another PR to fix the issue with choosing instructions.

Ah yes. I'm doing that

**Comment by tnkhanh** [comment]
Most were auto-merged. There were only 2 small conflicts in `src/llvm/llcompile.ml` (the instrument timing call to use `Sys.` ) and `llinstrument.ml` (the name change `Llcfg` -> `Llcallgraph`)

**Comment by taquangtrung** [comment]
> Most were auto-merged. There were only 2 small conflicts in `src/llvm/llcompile.ml` (the instrument timing call to use `Sys.` ) and `llinstrument.ml` (the name change `Llcfg` -> `Llcallgraph`)

Can compile successfully?

**Comment by tnkhanh** [comment]
> > Most were auto-merged. There were only 2 small conflicts in `src/llvm/llcompile.ml` (the instrument timing call to use `Sys.` ) and `llinstrument.ml` (the name change `Llcfg` -> `Llcallgraph`)
> 
> Can compile successfully?

Yes it compiles sucessfully

**Pull request merged**